bl/evaluation/evaluator.go

```diff
@@ -107,55 +116,112 @@ func (e *Evaluator) Evaluate(policyCheckData PolicyCheckData) (PolicyCheckResult
 	}
 
-	jsonSchemaValidator := jsonSchemaValidator.New()
+	emptyPolicyCheckResult := PolicyCheckResultData{FormattedResults{}, []cliClient.RuleData{}, []cliClient.FileData{}, nil, 0}
+
+	var filesData []cliClient.FileData
+	for _, filesConfiguration := range policyCheckData.FilesConfigurations {
+		filesData = append(filesData, cliClient.FileData{FilePath: filesConfiguration.FileName, ConfigurationsCount: len(filesConfiguration.Configurations)})
+	}
+
+	rulesData := []cliClient.RuleData{}
+	for _, rule := range policyCheckData.Policy.Rules {
+		rulesData = append(rulesData, cliClient.RuleData{Identifier: rule.RuleIdentifier, Name: rule.RuleName})
+	}
 
 	// map of files paths to map of rules to failed rule data
 	failedRulesByFiles := make(FailedRulesByFiles)
+	for _, filesConfiguration := range policyCheckData.FilesConfigurations {
+		for _, configuration := range filesConfiguration.Configurations {
+			err := e.evaluateConfiguration(failedRulesByFiles, policyCheckData, filesConfiguration.FileName, configuration)
+			if err != nil {
+				return emptyPolicyCheckResult, err
+			}
+		}
 
-	rulesData := []cliClient.RuleData{}
-	var filesData []cliClient.FileData
+	}
 
-	emptyPolicyCheckResult := PolicyCheckResultData{FormattedResults{}, []cliClient.RuleData{}, []cliClient.FileData{}, nil, 0}
+	formattedResults := FormattedResults{}
+	formattedResults.EvaluationResults = e.formatEvaluationResults(failedRulesByFiles, len(policyCheckData.FilesConfigurations))
 
-	for _, filesConfiguration := range policyCheckData.FilesConfigurations {
-		filesData = append(filesData, cliClient.FileData{FilePath: filesConfiguration.FileName, ConfigurationsCount: len(filesConfiguration.Configurations)})
+	if !policyCheckData.IsInteractiveMode {
+		formattedResults.NonInteractiveEvaluationResults = e.formatNonInteractiveEvaluationResults(formattedResults.EvaluationResults, failedRulesByFiles, policyCheckData.PolicyName, rulesCount)
+	}
 
-		for _, configuration := range filesConfiguration.Configurations {
-			for _, ruleWithSchema := range policyCheckData.Policy.Rules {
-				rulesData = append(rulesData, cliClient.RuleData{Identifier: ruleWithSchema.RuleIdentifier, Name: ruleWithSchema.RuleName})
+	return PolicyCheckResultData{formattedResults, rulesData, filesData, failedRulesByFiles, rulesCount}, nil
+}
 
-				configurationName, configurationKind := extractConfigurationInfo(configuration)
 
-				configurationJson, err := json.Marshal(configuration)
-				if err != nil {
-					return emptyPolicyCheckResult, err
-				}
 
-				ruleSchemaJson, err := json.Marshal(ruleWithSchema.Schema)
-				if err != nil {
-					return emptyPolicyCheckResult, err
-				}
+func (e *Evaluator) evaluateConfiguration(failedRulesByFiles FailedRulesByFiles, policyCheckData PolicyCheckData, fileName string, configuration extractor.Configuration) error {
+	skipAnnotations := extractSkipAnnotations(configuration)
+	if _, ok := skipAnnotations[SKIP_ALL_KEY]; ok {
+		// addSkipRule
```

Comment:
in case of skip all rules, we should skip `policyCheckData` or evaluateRule loop ? we well skip the policy check overall and 

> `>>  File: pkg/policy/tests/1-fail.yaml

> [V] YAML validation
> [V] Kubernetes schema validation
> [?] Policy check didn't run for this file
> 
> All rules are skipped for this resource because…
> “My skip reason”